### PR TITLE
Add gazebo Entity id to rendering sensor's user data

### PR DESCRIPTION
**Pull request by iche033**
Signed-off-by: Ian Chen <ichen@osrfoundation.org>

# 🎉 New feature

## Summary

Adds the `gazebo::Entity` id to a `rendering::Sensor`'s UserData object. The main use case is so that we can get back the corresponding gazebo `Entity` in the rendering thread when processing sensors.

## Test it
Run the `INTEGRATION_sensors_system` test.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/integration/sensors_system.cc [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 #include <string>
 #include <vector>
+#include <unordered_set>
```
Comment (ahcorde) [inline test/integration/sensors_system.cc:22]: alphabetize

**Review by ahcorde — CHANGES_REQUESTED** [review]
alphabetize headers

File: test/integration/sensors_system.cc (hunk 1 above)
Comment (iche033) [reply test/integration/sensors_system.cc:21]: 83432faec87bc75511089be690821f49970deae8

**Review by nkoenig — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**